test/config-luatest/credentials_applier_test.lua

```diff
@@ -560,3 +561,60 @@ g.test_set_password = function(g)
         child:close()
     end
 end
+
+g.test_remove_user_role = function(g)
+    -- Verify that when user or role is removed from the config,
+    -- it is not being deleted.
+
+    helpers.reload_success_case(g, {
+        options = {
+            credentials = {
+                roles = {
+                    myrole = {
+                        privileges = {{
+                            permissions = {
+                                'read',
+                                'write',
+                            },
+                            universe = true,
+                        }}
+                    },
+                },
+                users = {
+                    guest = {
+                        roles = { 'super' }
+                    },
+                    myuser = {
+                        privileges = {{
+                            permissions = {
+                                'execute',
+                            },
+                            universe = true,
+                        }},
+                    },
+                }
+            }
+        },
+        verify = function()
+            local ok, err = pcall(box.schema.user.info, 'myuser')
+            t.assert(ok, err)
+            ok, err = pcall(box.schema.role.info, 'myrole')
+            t.assert(ok, err)
+        end,
+        options_2 = {
+            credentials = {
+                users = {
+                    guest = {
+                        roles = { 'super' }
+                    },
+                }
+            }
+        },
+        verify_2 = function()
+            local ok, err = pcall(box.schema.user.info, 'myuser')
+            t.assert(ok, err)
+            ok, err = pcall(box.schema.role.info, 'myrole')
+            t.assert(ok, err)
```

Comment:
It looks worthful to verify that permissions that are granted on the first config apply are still there.

Reply:
Done. I've also moved verification function to a local var for convenience.